Parser/Node.cs

```diff
@@ -30,6 +30,12 @@ public record NVarDecl (Token Name, NType Type) : Node {
    public override string ToString () => $"{Type} {Name}";
 }
 
+// Declares a constant
+public record NConstDecl (Token Name, NExpr Value) : Node {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+   public override string ToString () => $"{Name} = {Value}";

```

Comment:
Why implement ToString only for this node? Either all nodes should have this, or we don't need it here either.